Apps/W1/HybridGP/app/src/Migration/Items/GPItemMigrator.codeunit.al

```diff
@@ -45,26 +48,40 @@ codeunit 4019 "GP Item Migrator"
     procedure OnMigrateItemPostingGroups(var Sender: Codeunit "Item Data Migration Facade"; RecordIdToMigrate: RecordId; ChartOfAccountsMigrated: Boolean)
     var
         GPItem: Record "GP Item";
-        GPItemLocation: Record "GP Item Location";
-        HelperFunctions: Codeunit "Helper Functions";
     begin
         if not ChartOfAccountsMigrated then
             exit;
 
         if RecordIdToMigrate.TableNo() <> Database::"GP Item" then
             exit;
 
-        Sender.CreateInventoryPostingSetupIfNeeded(CopyStr(PostingGroupCodeTxt, 1, 20), CopyStr(PostingGroupDescriptionTxt, 1, 50), '');
-        Sender.SetInventoryPostingSetupInventoryAccount(CopyStr(PostingGroupCodeTxt, 1, 20), '', HelperFunctions.GetPostingAccountNumber('InventoryAccount'));
-        if GPItemLocation.FindSet() then
-            repeat
-                Sender.CreateInventoryPostingSetupIfNeeded(CopyStr(PostingGroupCodeTxt, 1, 20), CopyStr(PostingGroupDescriptionTxt, 1, 50), CopyStr(GPItemLocation.LOCNCODE, 1, 10));
-                Sender.SetInventoryPostingSetupInventoryAccount(CopyStr(PostingGroupCodeTxt, 1, 20), CopyStr(GPItemLocation.LOCNCODE, 1, 10), HelperFunctions.GetPostingAccountNumber('InventoryAccount'));
-            until GPItemLocation.Next() = 0;
-
         if GPItem.Get(RecordIdToMigrate) then
-            if GPItem.ItemType = 0 then
-                Sender.SetInventoryPostingGroup(CopyStr(PostingGroupCodeTxt, 1, 20));
+            MigrateItemInventoryPostingGroup(GPItem, Sender);
+    end;
+
+    procedure MigrateItemInventoryPostingGroup(GPItem: Record "GP Item"; var Sender: Codeunit "Item Data Migration Facade")
+    var
+        GPIV00101: Record "GP IV00101";
+    begin
+        if GPCompanyAdditionalSettings.Name <> CompanyName() then begin
+            GPCompanyAdditionalSettings.GetSingleCompanyInstance(CompanyName());
+            GPCompanyAdditionalSettings."Migrate Item Classes" := true;
+            GPCompanyAdditionalSettings.Modify();
```

Comment:
Why are we setting company setting values here?

Reply:
Ahh, I didn't mean to check that in... Good catch.